pkg/scheduler/internal/cache/cache.go

```diff
@@ -693,17 +677,17 @@ func (cache *cacheImpl) addNodeImageStates(node *v1.Node, nodeInfo *framework.No
 			// update the entry in imageStates
 			state, ok := cache.imageStates[name]
 			if !ok {
-				state = &imageState{
-					size:  image.SizeBytes,
-					nodes: sets.New(node.Name),
+				state = &framework.ImageStateSummary{
+					Size:  image.SizeBytes,
+					Nodes: sets.New[string](node.Name),
 				}
 				cache.imageStates[name] = state
 			} else {
-				state.nodes.Insert(node.Name)
+				state.Nodes.Insert(node.Name)
 			}
-			// create the imageStateSummary for this image
+			// create the ImageStateSummary for this image
 			if _, ok := newSum[name]; !ok {
-				newSum[name] = cache.createImageStateSummary(state)
+				newSum[name] = state
```

Comment:
IIUC, the problem we refer to the same state across nodeInfos is that we expose the entrance to modify the cache in scheduling cycle. It may lead to data race.

Reply:
there is no race, scheduler use `Snapshot` in scheduling cycle,  the `Snapshot` is updated  `UpdateSnapshot` function  that generate copy of NodeInfo according to cache when beginning to schedule pod.

Reply:
Make sense, you added a new snapshot for ImageStateSummary. Thanks.